web/source/text/processor.ts

```diff
@@ -30,49 +30,84 @@ namespace com.keyman.text {
     // Denotes whether or not KMW needs to 'swallow' the next keypress.
     swallowKeypress: boolean = false;
 
-
     /**
-     * Get the default key string. If keyName is U_xxxxxx, use that Unicode codepoint.
-     * Otherwise, lookup the  virtual key code (physical keyboard mapping)
+     * Get the default RuleBehavior for the specified key, attempting to mimic standard browser defaults 
+     * where and when appropriate.
      *
      * @param   {object}  Lkc  The pre-analyzed key event object
      * @param   {number}  keyShiftState
      * @param   {boolean} usingOSK
      * @return  {string}
      */
-    defaultKeyOutput(Lkc: KeyEvent, keyShiftState: number, usingOSK: boolean): string {
-      let outputTarget = Lkc.Ltarg;
-
+    defaultRuleBehavior(Lkc: KeyEvent, keyShiftState: number, usingOSK: boolean): RuleBehavior {
       let keyman = com.keyman.singleton;
 
+      let outputTarget = Lkc.Ltarg;
+      let preInput = Mock.from(outputTarget);
+      let ruleBehavior = new RuleBehavior();
+
       // check if exact match to SHIFT's code.  Only the 'default' and 'shift' layers should have default key outputs.
       if (keyShiftState == Codes.modifierCodes['SHIFT']) {
```

Comment:
I wonder if we should put a TODO against this for future mnemonic support? Because mnemonic keyboards also have default output for AltGr, etc.

Reply:
I'll put that in defaultOutput.ts's implementation of `forBaseKeys`.